### PR TITLE
Fix mAsset Staking and improve usability

### PR DESCRIPTION
**Pull request by AstonKhor**
There was an issue with LCD servers running out of sync if they are queried too quickly. I've set the time to 15 seconds between requests so that the servers have time to be up synced before a request is made.

I also improved the output logging so it is more clear to the user what is happening.
Awesome program. Thanks for making it.

### CODE REVIEW

File: src/AutoStaker.ts [hunk 1]
```diff
@@ -63,6 +65,7 @@ export default class AutoStaker {
     }
 
     await this.pollingTx(result.txhash);
+    await sleep(15000); // necessary to stagger contract execution so that various LB'd servers can be synced on the last transaction completed
```
Comment (yun-yeo) [inline src/AutoStaker.ts]: Can we make this configurable??

File: src/AutoStaker.ts [hunk 2]
```diff
@@ -260,34 +250,54 @@ export default class AutoStaker {
...
 
     console.log('Done');
   }
+  async rewardsToClaim() {
+    const poolInfo = await this.mirror.staking.getPoolInfo(this.assetTokenAddr);
+    const rewardInfoResponse = await this.mirror.staking.getRewardInfo(
+      this.wallet.key.accAddress,
+      this.assetTokenAddr
+    );
+    const hasRewards = poolInfo.reward_index !== rewardInfoResponse.reward_infos[0].index;
```
Comment (yun-yeo) [inline src/AutoStaker.ts]: Also need to check pending_reward

**Review by yun-yeo — CHANGES_REQUESTED** [review]
Maybe need to solve conflict and #1 did some optimization like selling ratio. So can you look into this?

File: src/AutoStaker.ts (hunk 2 above)
Comment (AstonKhor) [reply src/AutoStaker.ts]: With the 15 second delay between contract executions there should never be pending rewards. Though I'm unfamiliar with how I might access info on pending_rewards.

File: src/AutoStaker.ts (hunk 1 above)
Comment (AstonKhor) [reply src/AutoStaker.ts]: Done.

**Comment by AstonKhor** [comment]
Alright, changes have been made. I don't feel we need to check pending_rewards, so I have left that unchanged, but please explain more if you think it'd be necessary.

**Review by yun-yeo — APPROVED** [review]

**Comment by yun-yeo** [comment]
Thanks for contribution :)

File: src/AutoStaker.ts (hunk 2 above)
Comment (yun-yeo) [reply src/AutoStaker.ts]: Understood

**Pull request merged**